pkg/controller/mongodb/mongodb_tls.go

```diff
@@ -201,6 +201,9 @@ func (r *ReplicaSetReconciler) completeTLSRollout(mdb mdbv1.MongoDB) error {
 
 	r.log.Debug("Completing TLS rollout")
 
+	if mdb.Annotations == nil {
+		mdb.Annotations = make(map[string]string)
+	}
```

Comment:
some tests were passing because they panicked here, and then got into the desired state after the operator was restarted

Reply:
good catch!